packages/ts-codegen/src/__tests__/index.test.ts

```diff
@@ -744,6 +744,310 @@ describe(generateTS, () => {
     });
   });
 
+  it("import an entity, which is used to implement interface and returned by resolver", () => {
+    let { resolvers, models } = runGenerateTest(graphql`
+      extend schema @import(from: "@msteams/packages-test", defs: ["Entity"])
+      
+      interface Person implements Entity {
+          id: ID!
+      }
+
+      type User implements Person {
+        id: ID!
+      }
+
+      extend type Query {
+        userById(id: ID!): Person
+      }
+    `);
+    expect(models).toMatchInlineSnapshot(`
+      "import type { EntityModel } from "@msteams/packages-test";
+      // Base type for all models. Enables automatic resolution of abstract GraphQL types (interfaces, unions)
+      export interface BaseModel {
+          __typename: string;
+      }
+      export interface PersonModel extends BaseModel, EntityModel {
+          __typename: string;
+      }
+      export interface UserModel extends BaseModel, PersonModel {
+          readonly __typename: "User";
+          readonly id: string;
+      }
+      "
+    `);
+    expect(resolvers).toMatchInlineSnapshot(`
+      "import type { PromiseOrValue } from "@graphitation/supermassive";
+      import type { ResolveInfo } from "@graphitation/supermassive";
+      import type { UserModel, PersonModel } from "./models.interface";
+      export declare namespace User {
+          export type id = (model: UserModel, args: {}, context: unknown, info: ResolveInfo) => PromiseOrValue<string>;
```

Comment:
Mostly out of curiosity; I don't believe there is any point in exporting items from within the namespace, or am I missing something?

Reply:
We need to be able to reach a field resolver within a type like `User.id` at the same time as `Conversation.id`. There was a conversation about this with you, I cannot quote, but you said something like that even though namespace is not recommended anymore and we should you module (based on the doc) this is the use-case which is an exception ?

Reply:
Yeah this was another topic. What I mean _here_ is that when you export a namespace, you don't also need to export the items within that exported namespace.